lib/wasm/src/environ/dummy.rs

```diff
@@ -52,6 +52,15 @@ pub struct DummyModuleInfo {
     /// Module and field names of imported functions as provided by `declare_func_import`.
     pub imported_funcs: Vec<(String, String)>,
 
+    /// Module and field names of imported globals as provided by `declare_global_import`.
+    pub imported_globals: Vec<(String, String)>,
+
+    /// Module and field names of imported tables as provided by `declare_table_import`.
+    pub imported_tables: Vec<(String, String)>,
+
+    /// Module and field names of imported memories as provided by `declare_memory_import`.
+    pub imported_memories: Vec<(String, String)>,
+
     /// Functions, imported and local.
     pub functions: PrimaryMap<FuncIndex, Exportable<SignatureIndex>>,
```

Comment:
Makes sense. Will do in a sec!

Reply:
(as a side note, I think there might be a better structure to define this import definitions, but I can handle that in other discussion / PR)

Reply:
@sunfishcode apart from creating the `DefinedTableIndex`, `DefinedGlobalIndex`, and `DefinedMemoryIndex` structures, how you think I should integrate into Dummy?

I'm a bit unsure about what you think we should add extra in `DummyEnvironment` (since `DefinedFuncIndex` is only used in `function_bodies` but we have nothing like this for tables, globals and memories).

Reply:
The dummy implementation may not need these, because it doesn't track everything. I guess one thing would be if we wanted to get rid of `GlobalInit::Import` and have implementations keep separate collections of global declarations and global definitions, similar to how we handle functions, but I don't have a strong opinion about that.

My main interest is in making the extra types available, for other environments which do record additional information for defined tables, and so on :-).

Reply:
Sounds good. Just added those :)